### PR TITLE
Randr update

### PR DESCRIPTION
**Pull request by mhhutton**
Updates Modal for submitting reviews. 
Post request works for posting data from review modal.
Code Clean up was done.
Still need to work on posting photos.
Sort function needs work.
Some of the validator functions need work.

### CODE REVIEW

File: client/src/components/reviews/AddReviewModal/AddReviewModal.jsx [hunk 1]
```diff
@@ -70,57 +133,265 @@ export default function ReviewDialog() {
...
+                type="email"
+                className="email"
+                helperText={validation.email}
+                error={validation.emailError}
+                onChange={(event) => setEmail(event.target.value)}
+              />
+              <Typography className="inputText">For authentication reasons, you will not be emailed</Typography>
+            </form>
 
+          </DialogContent>
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/AddReviewModal/AddReviewModal.jsx]: Could extract this to a different file for better separation of concerns

File: client/src/components/reviews/AddReviewModal/AddReviewModal.jsx [hunk 2]
```diff
@@ -49,19 +68,63 @@ const DialogTitle = withStyles(styles)((props) => {
...
+
+  // console.log('this is rating ', rating);
+  // console.log('this is recommend ', recommend);
+  // console.log('this is body', body);
+  // console.log('this is summary', summary);
+  // console.log('this is email', email);
+  // console.log('this is name', name);
+  // console.log('prod id', product.id);
+  // console.log('this is CHAR', characteristics);
+
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/AddReviewModal/AddReviewModal.jsx]: should clean up extra console logs

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
+import React from 'react';
+import clsx from 'clsx';
+import { makeStyles } from '@material-ui/core/styles';
+import { Radio, Typography } from '@material-ui/core';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormControl from '@material-ui/core/FormControl';
+import FormLabel from '@material-ui/core/FormLabel';
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx]: could consider condensing this a little more

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+export function buildCharRadios(charArr, setRecommend, setCharacteristics, characteristics) {
+  const handleCharacteristics = (event) => {
+    const resultObj = characteristics;
+    if (event.target.value.length > 0) {
+      const str = event.target.value;
+      const key = str.slice(2, str.length);
+      const val = Number(str.slice(0, 1));
+      console.log('SPLICE', key);
+      console.log('SPLICE2', val);
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx]: Should be removed for push to master

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+export function getAllCharVals(prodID) {
+  const resultsArr = [];
+  if (prodID === 18078) {
+    resultsArr.push(['60618', 'Fit', [
+      [1, 'Runs Tight'],
+      [2, 'Runs slightly tight'],
+      [3, 'Perfect'],
+      [4, 'Runs slightly Long'],
+      [5, 'Runs Long']
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx:74]: could consider converting fit to an object and just assign the ids later to increase dry code

File: client/src/components/reviews/ReviewTile/ReviewTile.jsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React, { useContext } from 'react';
+import React, { useContext} from 'react';
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/ReviewTile/ReviewTile.jsx]: Linting

File: server/routes/reviews.js [hunk 7]
```diff
@@ -37,6 +37,35 @@ router.get('/revs', (req, res) => {
...
+    body: req.body.body,
+    recommend: req.body.recommend,
+    name: req.body.name,
+    email: req.body.email,
+    photos: req.body.photos,
+    characteristics: req.body.characteristics
+  }, {
+    headers: {
+      Authorization: GITHUB_API_KEY,
+      'Content-Type': 'application/json'
```
Comment (Pix3lAssassin) [inline server/routes/reviews.js:54]: not sure this is necessary anymore

File: client/src/components/reviews/RatingsAndReviews.jsx [hunk 8]
```diff
@@ -19,9 +20,18 @@ const RatingsAndReviews = () => {
...
+    if (reviewTileList.length < 2) {
+      for (let i = 0; i < 2; i += 1) {
+        reviewTileList.unshift(reviewsArr[i]);
+      }
+    }
+  };
 
-  // getModalStyle is not a pure function, roll the style only on the first render
-  // const [open, setOpen] = useState(false);
+  // const [helpfulnessCount, setHelpfulnessCount] = useState(0);
```
Comment (ThinkCreatively) [inline client/src/components/reviews/RatingsAndReviews.jsx:34]: Do you need this commented out code?

**Review by ThinkCreatively — APPROVED** [review]

File: client/src/components/reviews/MoreReviewsBtn.jsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            setTileCount(res2 + 1);
+          }
+          setReviewTileList([...arr, ...reviewTileList]);
+        })
+        .then(() => {
+          if (lastVal === false) {
+            setReviewTileList([...arr, ...reviewTileList]);
+          }
+        });
+    }
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/MoreReviewsBtn.jsx:45]: Probably an easier way to do this

Comment (mhhutton) [reply client/src/components/reviews/MoreReviewsBtn.jsx:45]: This one was tricky, this was the first thing I got to work successfully.

File: client/src/components/reviews/ReviewTile/ReviewTile.jsx (hunk 6 above)
Comment (mhhutton) [reply client/src/components/reviews/ReviewTile/ReviewTile.jsx]: Fixed.

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx (hunk 5 above)
Comment (mhhutton) [reply client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx:74]: I tried using an object. I have difficulty extracting it in the value= within the FormControlLabel

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx (hunk 4 above)
Comment (mhhutton) [reply client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx]: Deleted.

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx (hunk 3 above)
Comment (mhhutton) [reply client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx]: Changed.

File: client/src/components/reviews/AddReviewModal/AddReviewModal.jsx (hunk 2 above)
Comment (mhhutton) [reply client/src/components/reviews/AddReviewModal/AddReviewModal.jsx]: Deleted.

File: client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx (hunk 5 above)
Comment (Pix3lAssassin) [reply client/src/components/reviews/AddReviewModal/modalHelperFunctions.jsx:74]: sry i mean define everything but the first index that way you can reuse the array just changing the id/first index

**Review by Pix3lAssassin — APPROVED** [review]

**Pull request merged**